packages/peregrine/src/Toasts/useToastActions.js

```diff
@@ -0,0 +1,54 @@
+import { useCallback } from 'react';
+import { useToastDispatch } from './context';
+
+// If a toast _is not_ dismissable remove it in this many milliseconds.
+const DEFAULT_TIMEOUT = 5000;
+
+/**
+ * Generates an identifier for a toast in a pure manner. Should always return
+ * the same identifier for a given set of props.
+ */
+const getToastId = props => {
+    return Object.keys(props).reduce((acc, curr) => acc + props[curr]);
```

Comment:
There's gotta be a better way to do this efficiently and consistently over string concatenation...